src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,131 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+        storageDao.add(new User("popova", "justpopova", 26));
+        storageDao.add(new User("sterniuk", "1123313", 30));
+        storageDao.add(new User("pony", "dfyrg6", 17));
+    }
+
+    @Test
+    public void userNull_isNotValid() {
+        Assertions.assertThrows(RuntimeException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    public void userSameLogin_isNotValid() {
+        User newUser = new User("popova", "justpopova", 26);
+        Assertions.assertThrows(RuntimeException.class, () -> {
+            registrationService.register(newUser);
+        });
+    }
+
+    @Test
+    public void userLogin_isValid() {
+        User newUser = new User("gabriella", "gabi2000", 22);
+        Assertions.assertEquals(newUser, registrationService.register(newUser));
+    }
+
+    @Test
+    public void userLoginNull_isNotValid() {
+        Assertions.assertThrows(RuntimeException.class, () -> {
+            registrationService.register(new User(null, "hellokitty", 19));
+        });
+    }
+
+    @Test
+    public void userLoginIsEmpty_isNotValid() {
+        String emptyName = "";
```

Comment:
follow this pattern: create test data - users with bad login or password etc, then do the action (register them), then compare results or expect exceptions